gunicorn/arbiter.py

```diff
@@ -167,6 +168,8 @@ def run(self):
         util._setproctitle("master [%s]" % self.proc_name)
 
         self.manage_workers()
+        if self.cfg.delay_pidfile and self.pidfile is not None:
+            self.pidfile.create(self.pid)
```

Comment:
I don't think this guarantee that the app has been fully initialized in the workers. AFAICT this will return as soon as all the worker processes have been spawned. They may still be initializing the app and not ready to accept requests.